powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/config/SignatureConfiguration.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2023 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.config;
+
+import io.getlime.security.powerauth.crypto.lib.enums.PowerAuthSignatureFormat;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Class that holds information about the signature configuration.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+public class SignatureConfiguration {
+
+    /**
+     * Key used for retrieving expected length of the signature component.
+     */
+    public static final String DECIMAL_SIGNATURE_COMPONENT_LENGTH = "DECIMAL_SIGNATURE_COMPONENT_LENGTH";
+
+    private final PowerAuthSignatureFormat signatureFormat;
+    private final Map<String, Object> signatureParameters = new LinkedHashMap<>();
```

Comment:
Just wondering, why do we not use strong typing (e.g. enumeration) for signature parameters? The signature parameters are known and very specific for each release, right? Or is there some deeper reason behind using a hashmap?

Reply:
I agree with this, the same issue for me.

Reply:
@hvge @romanstrobl Yes, there is a deeper reason. The type of signature is dynamically determined based on a signature header. We need to adjust the parameters later. Sure, if you can enumerate all possible use cases we will encounter in the next 5-10 releases, we can strongly type it.

Reply:
Fixed.